### ISSUE TITLE
cydia says "sub-process/usr/libexec/cydia/cydo returned an error code (2)" and can' proceed.

### ISSUE THREAD

**Issue by qunwang6** [checkra1n/BugTracker#942]
cydia says "sub-process/usr/libexec/cydia/cydo returned an error code (2)" and can' proceed.
iphone 6
iOS 12.4.4
iOS 12.4.5

❌ [[Question] After checkra1n, my cydia says "sub-process/usr/libexec/cydia/cydo returned an error code (2)" and can' proceed. : jailbreak](https://www.reddit.com/r/jailbreak/comments/duq71p/question_after_checkra1n_my_cydia_says/)
❌  [Sub - process /usr/libexec/cydia/cydo returned an error code (2) · Issue #119 · checkra1n/BugTracker](https://github.com/checkra1n/BugTracker/issues/119)
❌  [Error code Sub-process dpkg --set-selections returned an error code (2) · Issue #194 · checkra1n/BugTracker](https://github.com/checkra1n/BugTracker/issues/194)

**Comment by eglacias** [comment]
I have discovered on iOS 12.4, that that is caused when I haven’t done a restore rootFS with uncover before using check rain.  Also on that operating system only uncover 3.8 is compatible, but version four is not. They can both be used together only if uncovered is fully installed,  .mount_rw   Empty file put in your root directory, and then install check rain just to re break only.

**Comment by qunwang6** [comment]
new version fixed,thanks.

**Comment by fr33wi11** [comment]
I am having same issue on Ipad Air2. ios 13.3.1. with Checkra1n beta 0.9.8.2. . I even restored and re jb  . but cydia has same error 2 issue. Any thoughts ?

**Comment by r0my** [comment]
Hello! I have the same error now on my iphone 8, 13.6, jb with checkrain. 

I got this message after i removed a source from cydia manually from two files with filza. I have made a mistake to add an youtube url to my sources. I couldn't delete it in cydia so i deleted it manually. Reboot, jb and the error code appears when i try to update a tweak. 

I also can't delete a tweak.

Frustration now!

I there a solution for this cydo problem without loosing all installed tweaks?

Sorry for my bad english.

Romy
